lib/translate/common.cc

```diff
@@ -31,8 +29,89 @@ void DebugLog(const char* format, ...) {
   va_end(args);
 }
 
-void SetDebugCallback(void (*callback)(const char*)) {
+void SetDebugCallback(void (*callback)(const char *)) {
   debug_callback = callback;
 }
 
-}  // namespace translator
+StatusCode MakeScsiOpcode(uint8_t val, scsi_defs::OpCode &opcode) {
```

Comment:
This seems unnecessary. Can't we just static_cast between the two?

Reply:
https://github.com/googleinterns/scsi2nvme/pull/76#discussion_r445684566

but now that I think about it, the compiler would catch static cast errors. @oicirtap is there another advantage of this `MakeScsiOpcode` function besides enum bounds checking to justify leaving it in?

Reply:
I'm not a fan of having to pass in an OpCode value as a parameter and then having that be modified. 
Are we going to standardize which "default value" of the OpCode we pass in? However, I do like having a function that does error checking for opcode casting, so I'm not sure what's the best way to go.

Reply:
Thinking about this some more, I think an Opcode validation function is a really reasonable thing to have, so maybe lets keep this function. That said, the opcode should probably be validates upstream of individual command translation so we don't have to repeatedly implement logic related to calling the function.
Like @eyhhuang said, passing in the opcode just to return a status code feels a little odd. Maybe the function doesn't take in an opcode at all and simply returns a status code indicating if the Opcode is static castable?

Reply:
so that function would just be one if statement

Reply:
i'm going to use to `OpcodeToString()` function to check bounds and then simply `static_cast` in `raw_to_scsi()`. I think this is the simplest way  and it does bounds checking correctly. If we put casting into a function, we would have to check if that function returned a error or not anyway in `raw_to_scsi()`. I'd rather keep everything simple.